File: win-prog/ProgWinVB/Text and Fonts/UnderlinedText/UnderlinedText.vb
'-----------------------------------------------
' UnderlinedText.vb (c) 2002 by Charles Petzold
'-----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Text
Imports System.Windows.Forms

Class UnderlinedText
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New UnderlinedText())
    End Sub

    Sub New()
        Text = "Underlined Text Using HotkeyPrefix"
        Font = New Font("Times New Roman", 14)
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim str As String = "This is some &u&n&d&e&r&l&i&n&e&d text!"
        Dim strfmt As New StringFormat()
        strfmt.HotkeyPrefix = HotkeyPrefix.Show

        grfx.DrawString(str, Font, New SolidBrush(clr), 0, 0, strfmt)
    End Sub
End Class

File: win-prog/ProgWinCS/Text and Fonts/UnderlinedText/UnderlinedText.cs
//---------------------------------------------
// UnderlinedText.cs © 2001 by Charles Petzold
//---------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Text;
using System.Windows.Forms;

class UnderlinedText: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new UnderlinedText());
     }
     public UnderlinedText()
     {
          Text = "Underlined Text Using HotkeyPrefix";
          Font = new Font("Times New Roman", 14);
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          string       str    = "This is some &u&n&d&e&r&l&i&n&e&d text!";
          StringFormat strfmt = new StringFormat();

          strfmt.HotkeyPrefix = HotkeyPrefix.Show;

          grfx.DrawString(str, Font, new SolidBrush(clr), 0, 0, strfmt);
     }
}

